### ISSUE TITLE
File manager (like where is "this pc") opens after an app exits fullscreen

### ISSUE THREAD

**Issue by HHonzik** [valinet/ExplorerPatcher#2543]
OS: Win 11 22000.2538 (stuck on this out-of-date version forever)

Yeah, for example this happens when I press F11 and make this page fullscreen and after i interact with the window itself in fullscreen (like pressing keys) randomly file manager opens.
![About ExplorerPatcher](https://github.com/valinet/ExplorerPatcher/assets/87828740/44291bd0-ba95-4189-85fd-dbda4342b01e)

**Comment by OysterStew** [comment]
Ditto

**Comment by pyrates999** [comment]
Previous releases of windows 11 aren't usually tested against EP.  So I would try updating to windows 11 23h2.

**Comment by JoshuaCWebDeveloper** [comment]
I'm getting this same bug.

Windows 11, OS Build 22631.3296
ExplorerPatcher Version 22621.3296.64.1

`Register as a shell extension` is unchecked:

![image](https://github.com/valinet/ExplorerPatcher/assets/6536826/ecf446b7-dae3-4931-9b16-8dfa146fdc52)

**Comment by pyrates999** [comment]
Try enabling this option: https://windowsreport.com/file-explorer-folder-windows-task-manager/

**Comment by JoshuaCWebDeveloper** [comment]
Hi @pyrates999, I tried enabling that option, unfortunately, after enabling the option and restarting `explorer.exe`, the problem still persists. 

![image](https://github.com/valinet/ExplorerPatcher/assets/6536826/0a48f134-92d5-4722-98f4-9594ade41232)

**Comment by pyrates999** [comment]
pressing the windows key plus E on the keyboard also makes file explorer open.  Please check that your E key on your keyboard isn't stuck.

open up notepad and leave it open for a while and check if there is anything typed into it.

**Comment by JoshuaCWebDeveloper** [comment]
> pressing the windows key plus E on the keyboard also makes file explorer open. Please check that your E key on your keyboard isn't stuck.
> 
> open up notepad and leave it open for a while and check if there is anything typed into it.

Not sure how I'm supposed to interpret this comment. Let's keep this issue focused on trying to diagnose and fix the bug in ExplorerPatcher.

**Comment by pyrates999** [comment]
It's supposed to check if the keyboard is randomly causing the E key to think it is being pressed.

**Comment by JoshuaCWebDeveloper** [comment]
Given that the description of this issue pretty clearly states that this isn't random, many people have come forward with this issue, and, since I use my computer daily, I would notice if the keyboard were sending random E presses, I think we can safely rule that out.

**Comment by pyrates999** [comment]
Check using the console in explorer patcher.  Open up the properties application for explorer patcher, go to advanced, and click on enable console.  Then when a windows explorer window opens on its own, copy and paste the results from the EP console.

**Comment by JoshuaCWebDeveloper** [comment]
Unfortunately, the console doesn't output much when it happens. Here is the full output:

```
Running on Windows 11, OS Build 10.0.22631.3296.
Loaded symbols
Setup user32 functions done
CAppResolverCacheBuilder::_AddUserPinnedShortcutToStart() = 47430
ITrayUIHost = 3BD318
explorer.exe!TrayUI_CreateInstance() = 1A5DFC
Setup explorer functions done
Setup uxtheme functions done
[AC] rcMonitorAssignment = CC74
[AC] blockBegin = CC89
[AC] blockEnd = CD15
[AC] Patched!
[CC] rcMonitorAssignment = 3CB7FB
[CC] rcWorkAssignment = 3CB8C8
[CC] blockBegin = 3CB813
[CC] blockEnd = 3CB8E1
[CC] Patched!
[TC] rcMonitorAssignment = 3E4040
[TC] blockBegin = 3E4059
[TC] blockEnd = 3E4108
[TC] Patched!
[TV] firstCallPrep = 28A6A
[TV] firstCallCall = 28A74
[TV] Patched!
[HC] match1 = 1D60D
[HC] match2 = 1D6D4
[HC] writeAt = 1D2CE
[HC] cleanup = 1D6E5-1D6FA
[HC] Patched!
PenMenuSystemTrayManager::GetDynamicSystemTrayHeightForMonitor() = 449220
Setup twinui.pcshell functions done
Setup combase functions done
Setup twinui functions done
Setup stobject functions done
Setup bthprops functions done
Setup pnidui functions done
Setup shell32 functions done
Setup windows.storage functions done
[IME] Context menu patch status: 1
Setup inputswitch functions done
Setup peopleband functions done
Installed hooks.
Play startup sound thread...
Signal shell ready...
Open Start on monitor thread
Started "Signal shell ready" thread.
Started "Open Start on current monitor" thread.
EP Service Window thread
Initialized taskbar centering module.
Registered Win+A, Win+B, and Win+N
[Positioning] Added settings for monitor 000000002A9017C3 : 0
Progman: 1901544
Progman hook: 639898897
>>> Signal shell ready.
Setup sndvolsso functions done
Ended "Signal shell ready" thread.
[EnsureXAML] 47 ms.
Started "Play startup sound" thread.
Ended "Play startup sound" thread.
0 0
[Updates] Starting daemon.
[Updates] Configured update policy on this system: "Check for updates but let me choose whether to download and install them".
[Updates] Path to module: C:\WINDOWS\dxgi.dll
[Updates] Checking against hash "22622.3296.64.1.15361a1b13c6f570"
[Updates] Update URL: https://github.com/valinet/ExplorerPatcher/releases/latest/download/ep_setup.exe
[Updates] Hash of remote file is "22622.3296.64.1.15361a1b13c6f570" (valid).
[Updates] Local version obtained from hash is 22622.3296.64.1.
[Updates] No updates are available.
0 46000000
0 46000000
0 46000000
0 46000000
```

The last 4 lines show explorer windows opening, the first is one I opened myself, the other three happened when I exited a full screen app.

**Comment by pyrates999** [comment]
I think this is a bug in Windows and not related to Explorer Patcher, see this article: https://www.howtogeek.com/884624/youre-not-alone-windows-11-is-randomly-opening-file-explorer/

**Comment by JoshuaCWebDeveloper** [comment]
> I think this is a bug in Windows and not related to Explorer Patcher, see this article: https://www.howtogeek.com/884624/youre-not-alone-windows-11-is-randomly-opening-file-explorer/

That link describes a similar issue, but not the same issue. With this bug, the window opens in the background and does not steal focus. Additionally, this only started happening when I upgraded ExplorerPatcher from `22621.2861.62.2` to `22621.3296.64.1`, and uninstalling ExplorerPatcher resolves the bug.

**Comment by pyrates999** [comment]
Can you give a list of software that causes this to happen and list the steps involved?

**Comment by JoshuaCWebDeveloper** [comment]
At this point, I think any fullscreen application will cause it, I find it easiest to just press F11 to make my browser go fullscreen. It also seems more likely to happen if an explorer window is open and recently active.

1. Open an explorer window (not the default location)
2. Switch to your browser window
3. Make browser fullscreen
4. Exit fullscreen
5. Wait for 5 seconds
6. Repeat steps 1-5 several times
7. Wait for a few minutes
8. Repeat steps 1-7

**Comment by JoshuaCWebDeveloper** [comment]
> Does it happen in Chrome?

Yes.

```
Google Chrome | 122.0.6261.129 (Official Build) (64-bit) (cohort: Stable)
-- | --
Revision | f18a44fedeb29764b2b5336c120fdd90ef1a3f5c-refs/branch-heads/6261@{#1057}
OS | Windows 11 Version 23H2 (Build 22631.3296)
JavaScript | V8 12.2.281.22
User Agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36
```

**Comment by pyrates999** [comment]
Please update to the latest release of Chrome, 123.

**Comment by pyrates999** [comment]
Do you have any other software installed that modifies the UI of windows?  If so, please list them here.

**Comment by Ahmad2304** [comment]
Hi, I'm also having this issue

**Comment by pyrates999** [comment]
Your windows build is from July 9/2024: https://support.microsoft.com/en-au/topic/july-9-2024-kb5040442-os-builds-22621-3880-and-22631-3880-0864308e-61cc-413b-8194-0294331aba52

Please install the latest windows updates.

In the settings app, untick "Get the latest updates as soon as available" so that preview updates are not automatically installed.

All preview updates are then shown as download and install and are never automatically installed if you have the above unticked.

Preview updates are also not fully tested by Microsoft and thus could have bugs.

You should only be installing preview updates if you are a developer or are OK with running pre-release versions of EP.

**Comment by pyrates999** [comment]
does it happen if you uninstall EP?

**Comment by Ahmad2304** [comment]
No it stops happening when i uninstall EP

**Comment by pyrates999** [comment]
do you have any other software installed that modifies the windows ui?

**Comment by Ahmad2304** [comment]
No

**Comment by pyrates999** [comment]
Is it predictable of when it happens? or is it random?

**Comment by Ahmad2304** [comment]
Yes, just like OP said it happens after exiting any fullscreen app

**Comment by pyrates999** [comment]
Please wait for a developer to respond then.

In the mean time, check if you have **Register as shell extension** ticked in the EP properties application, and untick it.

**Comment by Ahmad2304** [comment]
No, I don't have it enabled.
![image](https://github.com/user-attachments/assets/de1c9266-4b02-4e3d-8f95-9687202417be)

**Comment by Traillam** [comment]
Hello, i have the exact same problem, i've try everything nothing work. It keep to happened everytime.

**Comment by Ahmad2304** [comment]
Try updating ExplorerPatcher it fixed it for me.
